psutil/__init__.py

```diff
@@ -1778,28 +1791,14 @@ def cpu_times_percent(interval=None, percpu=False):
 
     def calculate(t1, t2):
         nums = []
-        all_delta = _cpu_tot_time(t2) - _cpu_tot_time(t1)
-        for field in t1._fields:
-            field_delta = getattr(t2, field) - getattr(t1, field)
-            try:
-                field_perc = (100 * field_delta) / all_delta
-            except ZeroDivisionError:
-                field_perc = 0.0
+        times_delta = _cpu_times_deltas(t1, t2)
+        all_delta = _cpu_tot_time(times_delta)
+        scale = 100.0 / max(1, all_delta)
```

Comment:
I'm having troubles with understanding this part. Can you add a comment explaining what it does?

Reply:
I just moved the calculation of `100 / all_delta` that was inside the loop before to outside of the loop. So `(100 * field_delta) / all_delta` becomes `field_delta * scale` where `scale` is "pre-calculated". The `max` bit is to avoid ZeroDivisionError in the case of `all_delta=0`. In this case `field_delta` will also be 0 so the percentages will zero out anyway. I took this from `top.c` (https://gitlab.com/procps-ng/procps/blob/v3.3.12/top/top.c#L5083) but I can revert to the old logic if you prefer.